Add Option to Log Prestige Gains

AERobert opened the pull request:
Added the ability to log gains of basic prestige resources after resetting. Specifically, the gains of Plasmids, Anti-Plasmids, Dark Energy, Harmony Crystals, Artifacts, Supercoiled Plasmids, Servants, and Skilled Servants are listed. A new message category, "Prestige Gains," has been added to allow players to view only the prestige-related messages. As with Achievement messages, the new Prestige messages are not cleared after resetting, allowing players to view messages from previous resets. These messages are displayed in the same green color used for loot-related messages in battle and for rewards from the Easter and Halloween events. A new option has also been added to the settings tab to control whether these messages should be printed (this currently defaults to off).

Closes #660

A new function, `logPrestigeGains(reset, gains)`, was added along with calls in each reset-specific function. Most of the data comes from `calcPrestigeGains()`. The calculations for Servant and Skilled Servant gains use logic based on the code that determines the number of servants at the start of the game in `main.js`, which also includes gains from Path Finder and Overlord.

I had to add several new strings to `strings.json`: five for formatting the list of gained resources, two for the singular forms of "Servants" and "Skilled Servants," and a further two for the new setting. The new strings were not added to any of the non-English string files in this PR.

Note that any saves containing any of the new messages will break past game versions that do not implement the "Prestige" message category. Deleting the data in "lastMsg.prestige" from the save makes it compatible again, or just clicking the "Clear All" messages button.  Loading games from previous versions works fine, so I don't think this will affect average players.

Example messages (using current phrasing):
- Mutual Assured Destruction earned you 130 Plasmids.
- Ascension earned you 2225 Plasmids, 139 Phage, and 2.4 Harmony Crystals.
-  Apotheosis earned you 1 Plasmid and 64 Supercoiled Plasmids.
- Matrix earned you 2,056 Anti-Plasmids, 167 Phage, 3 Servants, and 1 Skilled Servant.  [after upgrading Path Finder and Overlord]

Attached screenshots of the above Ascension message in the log and one of the settings tab with the new option.

<img width="1470" height="956" alt="screenShot1_ascMessage" src="https://github.com/user-attachments/assets/3d30fb1e-9f55-466c-a37f-0b6131e29c0b" />

<img width="1470" height="956" alt="screenShot2_newSetting" src="https://github.com/user-attachments/assets/6a470617-ceef-49ee-834b-983887e2fe84" />

AERobert committed:
log basic prestige gains

The game now logs gains of basic resources after resetting. A new message category, "Prestige Gains," has been added to allow players to view only the prestige-related messages. As with Achievement messages, the new Prestige messages are not cleared after resetting, allowing players to view messages from previous resets. These messages are displayed in the same color as messages for loot gained in battle and rewards from the Easter and Halloween events (light green).

A new function `logPrestigeGains(reset)` was added along with calls in each reset specific function.

I had to add several new strings to strings.json, mainly for formatting the list of gained resources along with two containing singular forms of "Servants" and "Skilled Servants".

Example messages:
- Gained 130 Plasmids from Mutual Assured Destruction.
- Gained 2225 Plasmids, 139 Phage, and 2.4 Harmony Crystals from Ascension.
- Gained 1 Plasmid and 64 Supercoiled Plasmids from Apotheosis.
- Gained 2056 Anti-Plasmids, 167 Phage, 3 Servants, and 1 Skilled Servant from Matrix.

AERobert committed:
log basic prestige gains

The game now logs gains of basic resources after resetting. A new message category, "Prestige Gains," has been added to allow players to view only the prestige-related messages. As with Achievement messages, the new Prestige messages are not cleared after resetting, allowing players to view messages from previous resets. These messages are displayed in the same color as messages for loot gained in battle and rewards from the Easter and Halloween events (light green).

A new function `logPrestigeGains(reset)` was added along with calls in each reset specific function.

I had to add several new strings to strings.json, mainly for formatting the list of gained resources along with two containing singular forms of "Servants" and "Skilled Servants".

Example messages:
- Gained 130 Plasmids from Mutual Assured Destruction.
- Gained 2225 Plasmids, 139 Phage, and 2.4 Harmony Crystals from Ascension.
- Gained 1 Plasmid and 64 Supercoiled Plasmids from Apotheosis.
- Gained 2056 Anti-Plasmids, 167 Phage, 3 Servants, and 1 Skilled Servant from Matrix.

AERobert committed:
log servant gains plus refactoring

As the number of gained servants and skilled servants are not included in the data from calcPrestige, separate calculations had to be added using code based off of the block updating the counts of servants at the start of a game in main.js. New strings for singular "Servant" and "Skilled Servant" were also added.

Also refactored code to figure out the correct name for the current resource in the report to be a switch statement.

AERobert committed:
log servant gains plus refactoring

As the number of gained servants and skilled servants are not included in the data from calcPrestige, separate calculations had to be added using code based off of the block updating the counts of servants at the start of a game in main.js. New strings for singular "Servant" and "Skilled Servant" were also added.

Also refactored code to figure out the correct name for the current resource in the report to be a switch statement.

AERobert committed:
add new setting to control prestige logging

Adds a new control to the settings tab to control weather prestige summaries should be logged. The new prestige messages will only be sent if the option is enabled (by default its off). Put it by "disable special events" as that was the only option that seemed vaguely related to messages.

2026-07-07: Fix merge conflicts with latest working branch, in particular changed new setting to be "settings19".

AERobert committed:
add new setting to control prestige logging

Adds a new control to the settings tab to control weather prestige summaries should be logged. The new prestige messages will only be sent if the option is enabled (by default its off). Put it by "disable special events" as that was the only option that seemed vaguely related to messages.

LoveBodhi commented:
Format in Chinese is different

e.g. `通过飞升重置，您获得了 x 质粒、y 噬菌体和 z 和谐水晶。`

AERobert committed:
try making prestige messages easier to translate

Rephrased strings to have the reset type be at the start of the sentence. For example: `Ascension earned you 2225 Plasmids, 139 Phage, and 2.4 Harmony Crystals.` I just left `prestige_report_list_end` to hold a '.'.

Separated the string for " and" into two, one with the comma and one without. It is hoped that this will make it easier for languages which never have a comma (or other separator) before the and.

Also made some spacing at the top of code blocks slightly more consistant.

AERobert committed:
try making prestige messages easier to translate

Rephrased strings to have the reset type be at the start of the sentence. For example: `Ascension earned you 2225 Plasmids, 139 Phage, and 2.4 Harmony Crystals.` I just left `prestige_report_list_end` to hold a '.'.

Separated the string for " and" into two, one with the comma and one without. It is hoped that this will make it easier for languages which never have a comma (or other separator) before the and.

Also made some spacing at the top of code blocks slightly more consistant.

AERobert commented:
I have tweaked the strings and code to try to make them easier to translate.

Rephrased strings so that the reset type appears at the start of the sentence. Now, an English message might read, `Ascension earned you 2225 Plasmids, 139 Phage, and 2.4 Harmony Crystals.` I left `prestige_report_list_end` to hold a period ('.').

Also separated the string for " and" into two: one version with the comma and one without. This change is meant to make translation easier for languages that never use a comma (or other separator) before there equivlent of "and."

@LoveBodhi, does this format work better for a Chinese translation?

LoveBodhi commented:
> I have tweaked the strings and code to try to make them easier to translate.
> 
> Rephrased strings so that the reset type appears at the start of the sentence. Now, an English message might read, `Ascension earned you 2225 Plasmids, 139 Phage, and 2.4 Harmony Crystals.` I left `prestige_report_list_end` to hold a period ('.').
> 
> Also separated the string for " and" into two: one version with the comma and one without. This change is meant to make translation easier for languages that never use a comma (or other separator) before there equivlent of "and."
> 
> @LoveBodhi, does this format work better for a Chinese translation?

This time, now fine.

AERobert committed:
Update formula to get old servant counts

Now gets the number of skilled servants earned from prestige from `global.race.servants.sbase`, instead of `global.race.servants.smax`.

Use optional chaining operator in the retrieval so game won't crash if the `global.race.servants` object doesn't yet exist (I was testing with post TP4 saves where the key was already present).
